tests/bsim/bluetooth/mesh/src/test_suspend.c

```diff
@@ -270,15 +344,76 @@ static void test_tx_suspend_disable_resume(void)
 	/* Allow publishing to finish before suspending. */
 	k_sleep(K_MSEC(100));
 	ASSERT_OK(bt_mesh_suspend());
+}
+
+static void dut_gatt_body(bool disable_bt)
+{
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+	ASSERT_OK_MSG(bt_mesh_prov_enable(BT_MESH_PROV_GATT), "Failed to enable GATT provisioner");
+	dut_status = DUT_RUNNING;
+	/* Let the Tester observe pb gatt advertisements before provisioning. The node should
+	 * advertise pb gatt service with 100 msec (ext adv) or 1000msec (legacy adv) interval.
+	 */
+	k_sleep(K_MSEC(1800));
+
+	LOG_INF("Provisioning DUT");
+	bt_mesh_provision(test_net_key, 0, 0, 0, dut_cfg.addr, dut_cfg.dev_key);
+
+	/* Let the Tester observe proxy advertisements */
+	k_sleep(K_MSEC(6500));
+
+	/* Send a mesh message to notify Tester that DUT is about to be suspended. */
+	dut_status = DUT_SUSPENDED;
+	bt_mesh_test_send_over_adv(&dut_status, sizeof(enum dut_mesh_status));
+	k_sleep(K_MSEC(150));
```

Comment:
I see that `k_sleep(K_MSEC(150))` is placed behind all instances of `bt_mesh_test_send_over_adv`. Any reason to not put it inside the function it self? Or, if this sleep is to ensure that the message has been send by the adv layer, this could be handled by using a semaphore and the end_cb of  `struct bt_mesh_send_cb` inside `mesh_test.c`. I think that would have been a little bit cleaner if this is the intention.

Reply:
It's to ensure that the adv/message is sent before suspending DUT. It's only really relevant in the cases where we suspend just after sending, so I suggest we keep it as is. 

I can add a comment to this case to make it more clear?

Reply:
Ok, sure. I would argue that for a sync mechanism this would be relevant for a lot of usecases, but it is not critical.